vunit/vivado/vivado.py

```diff
@@ -81,18 +94,19 @@ def create_compile_order_file(project_file, compile_order_file, vivado_path=None
 
 def _read_compile_order(file_name, fail_on_non_hdl_files):
     """
-    Read the compile order file and filter out duplicate files
+    Read the compile order file and filter out duplicate files.
     """
     compile_order = []
     unique = set()
     include_dirs = set()
     libraries = set()
 
+    valid_file_types = ("Verilog", "VHDL", "Verilog Header", "SystemVerilog")
     with Path(file_name).open("r", encoding="utf-8") as ifile:
         for line in ifile.readlines():
             library_name, file_type, file_name = line.strip().split(",", 2)
 
-            if file_type not in ("Verilog", "VHDL", "Verilog Header"):
+            if file_type not in valid_file_types:
```

Comment:
I would use a list instead of a tuple.

Reply:
There lines fix #796

Reply:
Conceptually, I understand the tuples as an array of a fixed size, while a list is kind of expected to grow/reduce. Hence, I tend to use lists in other places of the codebase.

However, if there is a performance benefit, it is ok to use tuples here, and I will take it into account when touching other parts of the codebase. It's always nice to learn something!